agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -78,6 +81,222 @@ namespace beerocks {
+    } else if (ieee1905_1::eMediaTypeGroup::IEEE_1901 == media_type_group) {
+        // TODO: Not supported yet
+    } else if (ieee1905_1::eMediaTypeGroup::MoCA == media_type_group) {
+        // TODO: Not supported yet
+    } else {
+        media_type = ieee1905_1::eMediaType::UNKNONWN_MEDIA;
+        result     = true;
+    }
+
+    return result;
+}
+
+/**
+ * @brief Adds link metrics to response message.
+ *
+ * Gets link metrics of the interface between the reporter AL MAC and the neighbor AL MAC.
+ * Creates a Transmitter Link Metric TLV or a Receiver Link Metric TLV or both and adds them to
+ * the response message.
+ *
+ * @param[in] local_interface_name Name of the local interface.
+ * @param[in] media_type The underlying network technology of the connecting interface.
+ * @param[in] reporter_al_mac 1905.1 AL MAC address of the device that transmits the response message.
+ * @param[in] neighbor_al_mac 1905.1 AL MAC address of a neighbor of the receiving device.
+ * @param[in] link_metrics_type The link metrics type requested: TX, RX or both.
+ * @param[in, out] cmdu_tx CMDU response message where TLVs must be added to.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool add_link_metrics(const std::string &local_interface_name,
+                             ieee1905_1::eMediaType media_type, const sMacAddr &reporter_al_mac,
+                             const sMacAddr &neighbor_al_mac,
+                             ieee1905_1::eLinkMetricsType link_metrics_type,
+                             ieee1905_1::CmduMessageTx &cmdu_tx)
+{
+    /**
+     * Metrics information associated to the link between the local interface and the neighbor's interface.
+     */
+    sLinkMetrics link_metrics;
+
+    /**
+     * Link metrics collector to use to get link metrics in the connecting interface.
+     */
+    std::unique_ptr<link_metrics_collector> collector;
+
+    /**
+     * Get a link metrics collector suitable for the underlying network technology of the
+     * connecting interface.
+     * Collector choice depends on bits 15 to 8 of media type.
+     */
+    uint8_t media_type_msb = media_type >> 8;
+    if (0 == media_type_msb) {
+        collector = std::make_unique<ieee802_3_link_metrics_collector>();
+    } else if (1 == media_type_msb) {
+        // TODO: Create a collector for wireless interfaces
+        //collector = std::make_unique<ieee802_11_link_metrics_collector>();
+    } else {
+        LOG(ERROR) << "Unsupported media type: " << std::hex << (int)media_type;
+        return (false);
+    }
+
+    /**
+     * Get link metrics information
+     */
+    if (collector &&
+        collector->get_link_metrics(local_interface_name, neighbor_al_mac, link_metrics)) {
+
+        /**
+         * Get local interface information to obtain its MAC address.
+         */
+        network_utils::iface_info iface_info;
+        if (network_utils::get_iface_info(iface_info, local_interface_name) != 0) {
```

Comment:
You only use the mac address, so better to use:
```suggestion
        std::string mac;
        if (!network_utils::linux_iface_get_mac(local_interface_name, mac)) {
```

Reply:
Done